assignment3/word_search.py

```diff
@@ -0,0 +1,63 @@
+class Word_searcher():
+    def __init__(self, rows, columns, matrix, words):
+        self.words = words
+        self.rows = rows
+        self.columns = columns
+        self.matrix = matrix
+        self.starting_points = {}
+        self.found_words = []
+
+    def _check_neighbours(self, x, y, letter):
```

Comment:
Why does the function name starts with underscore? 
- _single_leading_underscore: weak "internal use" indicator.  E.g. "from M
  import *" does not import objects whose name starts with an underscore.